src/Reference/HasOne.php

```diff
@@ -228,18 +236,14 @@ public function ref($defaults = []) : Model
                 $m->addHook('afterSave', function ($m) {
                     $this->owner[$this->our_field] = $m[$this->their_field];
                 });
-        } else {
-            if ($this->owner[$this->our_field]) {
-                $m->tryLoad($this->owner[$this->our_field]);
-            }
-
-            return
-                $m->addHook('afterSave', function ($m) {
-                    $this->owner[$this->our_field] = $m->id;
-                });
+        }
+        if ($this->owner[$this->our_field]) {
```

Comment:
What if `$this->our_field` is not defined at all (equals `null`)? I think then it will throw exception here.
I know it was like that before too, but probably should be `if ($this->our_field !== null && $this->owner[$this->our_field])` here to be 100% sure it's set.

Reply:
We can put a conditional, but there was no need. probably because reference is never called directly, but from Model and is always associated with a field_name which in relations became our_field, i'm wrong?

Reply:
ok fine